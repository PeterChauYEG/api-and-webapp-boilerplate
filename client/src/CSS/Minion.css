.Minion {
  width: 75%;
  height: 100%;
  border-bottom: 1px dashed #333333;
}