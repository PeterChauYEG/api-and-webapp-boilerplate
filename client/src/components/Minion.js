import React from 'react'

// import css
import '../CSS/Minion.css'

const Minion = (props) => {
  const {
    minion: {
      brand,
      description,
      name,
    },
  } = props;

  return (
    <div className="Minion">
      <h2>{name}</h2>
      <h3>{brand}</h3>
      <p>{description}</p>
    </div>
  );
}

Minion.propTypes = {
  minion: React.PropTypes.object.isRequired,
}

export default Minion;
