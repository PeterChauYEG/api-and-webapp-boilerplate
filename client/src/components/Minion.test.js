import React from 'react'

// import test dependencies 
import renderer from 'react-test-renderer'

// import component
import Minion from './Minion'

// render test
test('renders without crashing', () => {
  
  // declare prop
  const minion = {
    brand: "brand",
    description: "description",
    name: "name",
  }
  
  // render component
  const component = renderer.create(
    <Minion 
      minion={minion}
      />
  )
  
  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});