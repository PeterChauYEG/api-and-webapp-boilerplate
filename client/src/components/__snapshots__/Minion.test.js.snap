exports[`test renders without crashing 1`] = `
<div
  className="Minion">
  <h2>
    name
  </h2>
  <h3>
    brand
  </h3>
  <p>
    description
  </p>
</div>
`;
